cmd/monaco/integrationtest/v2/slo-v2_integration_test.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/coordinate"
+	manifestloader "github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/manifest/loader"
+)
+
+func TestSloV2(t *testing.T) {
+	configFolder := "test-resources/slo-v2/"
+	manifestPath := configFolder + "manifest.yaml"
+	environment := "platform_env"
+	project := "project"
+
+	// enable FF
+	t.Setenv(featureflags.ServiceLevelObjective.EnvName(), "true")
+
+	t.Run("When deploying two configs, two configs exist", func(t *testing.T) {
+		RunIntegrationWithCleanup(t, configFolder, manifestPath, environment, "SLO-V2", func(fs afero.Fs, testContext TestContext) {
+			err := monaco.RunWithFSf(fs, "monaco deploy %s --project=%s --verbose", manifestPath, project)
+			assert.NoError(t, err)
+
+			err = monaco.RunWithFSf(fs, "monaco deploy %s --project=%s --verbose", manifestPath, project)
+			assert.NoError(t, err)
+
+			sloV2Client := createSloV2Client(t, fs, manifestPath, environment)
+			result, err := sloV2Client.List(t.Context())
+			assert.NoError(t, err)
+			externalIDs := parseExternalIDs(t, result)
+
+			cSliCoord := coordinate.Coordinate{
+				Project:  project,
+				Type:     string(config.ServiceLevelObjectiveID),
+				ConfigId: "custom-sli_" + testContext.suffix,
+			}
+			sliRefCoord := coordinate.Coordinate{
+				Project:  project,
+				Type:     string(config.ServiceLevelObjectiveID),
+				ConfigId: "sli-reference_" + testContext.suffix,
+			}
+
+			cSliExternalID := idutils.GenerateExternalID(cSliCoord)
+			sliRefExternalID := idutils.GenerateExternalID(sliRefCoord)
+
+			assert.Contains(t, externalIDs, cSliExternalID)
+			assert.Contains(t, externalIDs, sliRefExternalID)
+		})
+	})
+
+	t.Run("With a disabled FF the deploy should fail", func(t *testing.T) {
+		t.Setenv(featureflags.ServiceLevelObjective.EnvName(), "false")
+
+		RunIntegrationWithoutCleanup(t, configFolder, manifestPath, environment, "SLO-V2", func(fs afero.Fs, testContext TestContext) {
+			// when deploying once
+			err := monaco.RunWithFSf(fs, "monaco deploy %s --project=%s --verbose", manifestPath, project)
+			assert.Error(t, err)
+
+			sloV2Client := createSloV2Client(t, fs, manifestPath, environment)
+			result, err := sloV2Client.List(t.Context())
+			assert.NoError(t, err)
+			externalIDs := parseExternalIDs(t, result)
+
+			coord := coordinate.Coordinate{
+				Project:  project,
+				Type:     string(config.ServiceLevelObjectiveID),
+				ConfigId: "custom-sli_" + testContext.suffix,
+			}
+			externalID := idutils.GenerateExternalID(coord)
+			assert.NotContains(t, externalIDs, externalID)
+		})
+	})
+}
+
+func parseExternalIDs(t *testing.T, response api.PagedListResponse) []string {
+	allObjects := response.All()
+	externalIds := make([]string, 0, len(allObjects))
+	for _, obj := range allObjects {
+		externalIds = append(externalIds, parseSloV2Payload(t, obj).ExternalId)
```

Comment:
Since you only use the `ExternalId` at all, maybe `parseSlo..` should return it directly. E.g. `extractSloV2ExternalId(t, obj)`.